port/src/devicetree.rs

```diff
@@ -0,0 +1,738 @@
+use core::ffi::CStr;
+
+#[derive(Debug)]
+pub enum ParseError {
+    InvalidHeader,
+    InvalidMagic,
+    BufferTooSmall,
+    InvalidToken,
+}
+
+type Result<T> = core::result::Result<T, ParseError>;
+
+fn bytes_to_u32(bytes: &[u8]) -> Option<u32> {
+    Some(u32::from_be_bytes(bytes.get(..4)?.try_into().unwrap()))
+}
+
+fn bytes_to_u64(bytes: &[u8]) -> Option<u64> {
+    Some(u64::from_be_bytes(bytes.get(..8)?.try_into().unwrap()))
+}
+
+/// Extract u32 from bytes, but cast as u64
+fn bytes_to_u32_as_u64(bytes: &[u8]) -> Option<u64> {
+    Some(u32::from_be_bytes(bytes.get(..4)?.try_into().unwrap()) as u64)
```

Comment:
Since you're going from u32 to u64, you _should_ just be able to call `.into()` on the result after the `unwrap`.